src/net.cpp

```diff
@@ -563,6 +563,9 @@ void CNode::Ban(const CSubNet& subNet, const BanReason &banReason, int64_t banti
         setBanned[subNet] = banEntry;
 
     setBannedIsDirty = true;
+
+    DumpBanlist(); //store banlist to disk
```

Comment:
Is it necessary to write to disk every time a ban comes in?  We can update the UI without writing to disk and having the `setBannedIsDirty` flag will catch it the next time `DumpData` gets hit (granted that's on a 15 minute interval).  My main concern here is what if we suddenly need to ban lots of nodes say during a DoS attack, we then end up writing to disk for each node banned (i.e. 50 bans = 50 writes where each write only adds one additional ban to the list).